spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialMaxPooling.scala

```diff
@@ -69,169 +72,288 @@ class SpatialMaxPooling[T: ClassTag](
     this
   }
 
+  // return (padTop, padDown, padLeft, padRight)
+  protected def getPadding(inputHeight: Int, inputWidth: Int): (Int, Int, Int, Int) = {
```

Comment:
I remember the same method appears in the conv2D layer. Can we make it as a utility function?